lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -161,11 +183,74 @@ def redcap_det_uw_reopening(*, db: DatabaseSession, cache: TTLCache, det: dict,
+            'odegaard': 'UWOdegaardLibrary',
+            'slu':      'UWSouthLakeUnion',
+            'tacoma':   'UWTacoma',
+            'uw_club':  'UWClub'
+            }
+
+        site_reference = create_site_reference(
+            location = record_location,
+            site_map = location_site_map,
+            default_site = SWAB_AND_SEND_SITE,
+            system_identifier = INTERNAL_SYSTEM)
+
+        # Handle various symptoms.
+        contained: List[dict]
+        diagnosis: List[dict]
+
+        # Map the various symptoms variables to their onset date.
+        # For daily_symptoms_covid_like we don't know the actual onset date. The questions asks
+        # "in the past 24 hours"
+        if event_type == EventType.ENCOUNTER:
+            symptom_onset_map = {
+                'daily_symptoms_covid_like': None,
+                'symptoms': redcap_record_instance['symptom_onset'],
+                'symptoms_kiosk': redcap_record_instance['symptom_duration_kiosk'],
+                'symptoms_swabsend': redcap_record_instance['symptom_duration_swabsend']
+            }
+        elif event_type == EventType.ENROLLMENT:
+            symptom_onset_map = {'symptoms_base': redcap_record_instance['symptom_onset_base']}
+
+        contained, diagnosis = build_contained_and_diagnosis(
+            patient_reference = patient_reference,
+            record = redcap_record_instance,
+            symptom_onset_map = symptom_onset_map,
+            system_identifier = INTERNAL_SYSTEM)
+
+        collection_code = None
+        if event_type == EventType.ENROLLMENT or collection_method == CollectionMethod.SWAB_AND_SEND:
+            collection_code = CollectionCode.HOME_HEALTH
+        elif collection_method == CollectionMethod.KIOSK:
+            collection_code = CollectionCode.FIELD
+
+        encounter_date = get_encounter_date(redcap_record_instance, event_type)
 
         initial_encounter_entry, initial_encounter_reference = create_encounter(
-            redcap_record_instance, patient_reference, site_reference,
-            location_resource_entries, event_type, collection_method)
+            encounter_date = encounter_date,
+            patient_reference = patient_reference,
+            site_reference = site_reference,
+            locations = location_resource_entries,
+            diagnosis = diagnosis,
+            contained = contained,
+            collection_code = collection_code,
+            parent_encounter_reference = None,
+            encounter_reason_code = None,
+            encounter_identifier_suffix = None,
+            system_identifier = INTERNAL_SYSTEM,
+            redcap_uri = create_redcap_uri(redcap_record_instance),
+            redcap_url = REDCAP_URL,
+            redcap_project_id = project_id,
+            redcap_record_id = record_id,
+            redcap_event_name = redcap_record_instance['redcap_event_name'],
+            redcap_repeat_instance = redcap_record_instance['redcap_repeat_instance'])
```

Comment:
This is a long list of arguments!!
I wonder if there's a data structure or a class we can use to improve this eventually.